packages/lmi/src/lmi/rate_limiter.py

```diff
@@ -240,10 +255,15 @@ async def get_rate_limit_keys(
         self, cursor_scan_count: int = 100
     ) -> list[tuple[RateLimitItem, tuple[str, str | MatchAllInputs]]]:
         """Returns a list of current RateLimitItems with tuples of namespace and primary key."""
-        host, port = os.environ.get("REDIS_URL", ":").split(":", maxsplit=2)
+        redis_url = self.redis_url or os.environ.get("REDIS_URL", ":")
+        if not redis_url:
+            raise ValueError(
+                "Missing Redis URL, pass at initialization or set env variable REDIS_URL"
+            )
+        host, port = redis_url.split(":", maxsplit=2)
```

Comment:
```suggestion
        try:
            host, port = redis_url.split(":", maxsplit=2)
        except ValueError as exc:
            raise ValueError(
                f"Failed to parse host and port from Redis URL {redis_url!r},"
                " correctly pass at initialization or set env variable REDIS_URL."
            ) from exc
```

The `not redis_url` check feels weird as we have `os.environ.get("REDIS_URL", ":")`, so it's a low probability of hitting this. This generalizes the error a bit